### PR TITLE
[FIX] Properly set number fo permutation chunks when n_jobs > n_perm

### LINKED ISSUE #4910: [BUG] doc build failure on main for several examples
All of them point to nilearn/mass_univariate/permuted_least_squares.py

I believe that this must be due to https://github.com/nilearn/nilearn/pull/4875

But the fact that our test suite did not catch this means we have a blind spot somewhere.

##  02_decoding/plot_oasis_vbm.py

https://github.com/nilearn/nilearn/actions/runs/12201002128/job/34077197805#step:13:27267

```
    Traceback (most recent call last):
      File "/home/runner/work/nilearn/nilearn/examples/02_decoding/plot_oasis_vbm.py", line 196, in <module>
        neg_log_pvals, t_scores_original_data, _ = permuted_ols(
      File "/home/runner/work/nilearn/nilearn/.tox/doc/lib/python3.9/site-packages/nilearn/mass_univariate/permuted_least_squares.py", line 781, in permuted_ols
        ret = joblib.Parallel(n_jobs=n_jobs, verbose=verbose)(
      File "/home/runner/work/nilearn/nilearn/.tox/doc/lib/python3.9/site-packages/joblib/parallel.py", line 1098, in __call__
        self.retrieve()
      File "/home/runner/work/nilearn/nilearn/.tox/doc/lib/python3.9/site-packages/joblib/parallel.py", line 975, in retrieve
        self._output.extend(job.get(timeout=self.timeout))
      File "/home/runner/work/nilearn/nilearn/.tox/doc/lib/python3.9/site-packages/joblib/_parallel_backends.py", line 567, in wrap_future_result
        return future.result(timeout=timeout)
      File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/concurrent/futures/_base.py", line 446, in result
        return self.__get_result()
      File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
        raise self._exception
    _pickle.PicklingError: Could not pickle the task to send it to the workers.
```

## 05_glm_second_level/plot_second_level_association_test.py

https://github.com/nilearn/nilearn/actions/runs/12201002128/job/34077197805#step:13:27286

```
    Traceback (most recent call last):
      File "/home/runner/work/nilearn/nilearn/examples/05_glm_second_level/plot_second_level_association_test.py", line 137, in <module>
        neg_log_pvals_permuted_ols_unmasked = non_parametric_inference(
      File "/home/runner/work/nilearn/nilearn/.tox/doc/lib/python3.9/site-packages/nilearn/glm/second_level/second_level.py", line 1119, in non_parametric_inference
        outputs = permuted_ols(
      File "/home/runner/work/nilearn/nilearn/.tox/doc/lib/python3.9/site-packages/nilearn/mass_univariate/permuted_least_squares.py", line 781, in permuted_ols
        ret = joblib.Parallel(n_jobs=n_jobs, verbose=verbose)(
      File "/home/runner/work/nilearn/nilearn/.tox/doc/lib/python3.9/site-packages/joblib/parallel.py", line 1098, in __call__
        self.retrieve()
      File "/home/runner/work/nilearn/nilearn/.tox/doc/lib/python3.9/site-packages/joblib/parallel.py", line 975, in retrieve
        self._output.extend(job.get(timeout=self.timeout))
      File "/home/runner/work/nilearn/nilearn/.tox/doc/lib/python3.9/site-packages/joblib/_parallel_backends.py", line 567, in wrap_future_result
        return future.result(timeout=timeout)
      File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/concurrent/futures/_base.py", line 439, in result
        return self.__get_result()
      File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
        raise self._exception
    _pickle.PicklingError: Could not pickle the task to send it to the workers.
```

## 07_advanced/plot_haxby_mass_univariate.py

https://github.com/nilearn/nilearn/actions/runs/12201002128/job/34077197805#step:13:27307

```
     Traceback (most recent call last):
      File "/home/runner/work/nilearn/nilearn/examples/07_advanced/plot_haxby_mass_univariate.py", line 104, in <module>
        neg_log_pvals, t_scores_original_data, _ = permuted_ols(
      File "/home/runner/work/nilearn/nilearn/.tox/doc/lib/python3.9/site-packages/nilearn/mass_univariate/permuted_least_squares.py", line 781, in permuted_ols
        ret = joblib.Parallel(n_jobs=n_jobs, verbose=verbose)(
      File "/home/runner/work/nilearn/nilearn/.tox/doc/lib/python3.9/site-packages/joblib/parallel.py", line 1098, in __call__
        self.retrieve()
      File "/home/runner/work/nilearn/nilearn/.tox/doc/lib/python3.9/site-packages/joblib/parallel.py", line 975, in retrieve
        self._output.extend(job.get(timeout=self.timeout))
      File "/home/runner/work/nilearn/nilearn/.tox/doc/lib/python3.9/site-packages/joblib/_parallel_backends.py", line 567, in wrap_future_result
        return future.result(timeout=timeout)
      File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/concurrent/futures/_base.py", line 439, in result
        return self.__get_result()
      File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
        raise self._exception
    _pickle.PicklingError: Could not pickle the task to send it to the workers.
```

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4910

Changes proposed in this pull request:

- revert one of the changes introduced by #4875
- add a regression test
- small changes in the failing examples so they are run in CI

## unrelated

- fixes linkchecks that's also failing on main

### CODE REVIEW

File: examples/07_advanced/plot_haxby_mass_univariate.py [hunk 1]
```diff
@@ -122,7 +122,8 @@
 
 # f_regression implicitly adds intercept
 _, pvals_bonferroni = f_regression(
-    grouped_fmri_masked, grouped_conditions_encoded
+    grouped_fmri_masked,
+    grouped_conditions_encoded.ravel(),
```
Comment (Remi-Gau) [inline examples/07_advanced/plot_haxby_mass_univariate.py:126]: unrelated but removes this sklearn warning

image

File: nilearn/mass_univariate/permuted_least_squares.py [hunk 2]
```diff
@@ -762,20 +762,23 @@ def permuted_ols(
...
+            f"The specified number of permutations is {n_perm} "
+            "and the number of jobs to be performed in parallel "
+            f"has set to {n_jobs}. "
+            f"This is incompatible so only {n_perm} jobs will be running. "
+            "You may want to perform more permutations "
+            "in order to take the most of the available computing resources.",
+            UserWarning,
+            stacklevel=2,
+        )
+        n_perm_chunks = np.ones(n_perm, dtype=int)
```
Comment (Remi-Gau) [inline nilearn/mass_univariate/permuted_least_squares.py:776]: before this PR this `n_perm_chunks = np.ones(n_perm, dtype=int)` was always ran

Comment (Remi-Gau) [reply nilearn/mass_univariate/permuted_least_squares.py:776]: so that was the problem

**Comment by Remi-Gau** [comment]
the regression test fails before the change and passes after

File: nilearn/mass_univariate/tests/test_permuted_least_squares.py [hunk 3]
```diff
@@ -768,6 +768,31 @@ def test_sanitize_inputs_permuted_ols(design):
     )
 
 
+def test_permuted_ols_warnings_n_perm_n_job(cluster_level_design, masker):
+    """Check that proper warning are thrown dependning on n_job VS n_perm."""
```
Comment (bthirion) [inline nilearn/mass_univariate/tests/test_permuted_least_squares.py:772]:
```suggestion
    """Check that proper warning are thrown depending on n_job VS n_perm."""
```

File: nilearn/mass_univariate/tests/test_permuted_least_squares.py [hunk 4]
```diff
@@ -768,6 +768,31 @@ def test_sanitize_inputs_permuted_ols(design):
...
+            target_var,
+            n_perm=4,
+            n_jobs=1,
+            masker=masker,
+        )
+    assert not any(
+        "perform more permutations" in str(x.message) for x in record
+    )
+
+    # n_perm <= n_job  and n_job > 0 --> no warning
```
Comment (bthirion) [inline nilearn/mass_univariate/tests/test_permuted_least_squares.py:788]:
```suggestion
    # n_perm <= n_job  and n_job > 0 -->  warning
```

**Review by bthirion — COMMENTED** [review]
Looks good overall. Bets,

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Review by man-shu — APPROVED** [review]
LGTM!

**Pull request merged**